build.sbt

```diff
@@ -1,67 +1,98 @@
+    name := "xenomorph",
+    publish := {},
+    publishLocal := {}
   )
 
-lazy val core = project.in(file("modules/core"))
+lazy val core = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("modules/core"))
   .settings(commonSettings)
   .settings(
     name := "xenomorph-core",
     libraryDependencies ++= Seq(
-      "org.scalaz"                    %% "scalaz-core"     % "7.2.14",
-      "com.chuusai"                   %% "shapeless"       % "2.3.2",
-      "com.github.julien-truffaut"    %% "monocle-core"    % monocleVersion,
+      "org.scalaz"                    %%% "scalaz-core"     % "7.2.18",
+      "com.chuusai"                   %%% "shapeless"       % "2.3.2",
+      "com.github.julien-truffaut"    %%% "monocle-core"    % monocleVersion,
       // test dependencies
-      "com.github.julien-truffaut"  %%  "monocle-generic"  % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-macro"    % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-state"    % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-refined"  % monocleVersion % "test",
-      "com.github.julien-truffaut"  %%  "monocle-law"      % monocleVersion % "test",
-      "joda-time"                   % "joda-time"          % "2.9.4"        % "test",
-      "org.specs2"                  %% "specs2-core"       % "3.9.4"        % "test",
-      "org.specs2"                  %% "specs2-scalacheck" % "3.9.4"        % "test"
+      "com.github.julien-truffaut"  %%%  "monocle-generic"  % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-macro"    % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-state"    % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-refined"  % monocleVersion % "test",
+      "com.github.julien-truffaut"  %%%  "monocle-law"      % monocleVersion % "test",
+      "org.scalatest"               %%% "scalatest"         % "3.0.4"        % "test",
+      "org.scalacheck"              %%% "scalacheck"        % "1.13.4"       % "test"
+    )
+  )
+  .jsSettings(
+    libraryDependencies ++= Seq(
+      "org.scala-js" %%% "scalajs-java-time" % "0.2.3" % "test"
     )
   )
 
-lazy val argonaut = project.in(file("modules/argonaut"))
+lazy val coreJVM = core.jvm
+lazy val coreJS = core.js
+
+lazy val argonaut = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("modules/argonaut"))
   .dependsOn(core % "compile->compile;test->test", scalacheck % "test->test")
   .settings(commonSettings)
   .settings(
     name := "xenomorph-argonaut",
     libraryDependencies ++= Seq(
-      "io.argonaut" %% "argonaut" % "6.2"
+      "io.argonaut" %%% "argonaut" % "6.2.1"
     )
   )
 
-lazy val scodec = project.in(file("modules/scodec"))
+lazy val argonautJVM = argonaut.jvm
+lazy val argonautJS = argonaut.js
+
+lazy val scodec = crossProject(JVMPlatform)
```

Comment:
sadly scalaz concurrent is not built against scalajs, preventing this module to cross compile. Still used the crossProject for consistency

Reply:
Interesting. I wonder if scodec depends upon scalaz.concurrent just for Task, in order to have a capable IO type? Although, I'd expect that scodec has other JVM-specific dependencies as well.

Reply:
After a quick look, it appears that the scodec-scalaz interop lib is the cause : I think this [right there](https://github.com/scodec/scodec-scalaz/blob/series/1.4.x/src/main/scala/scodec/interop/scalaz/package.scala#L26) could be the only the lib makes of `scalaz-concurrent`. 

That library is only 2 source files, and I think its only use here is to have a `Monad[scodec.Attempt]`. Would be worth directly implementing it in `xenomorph` and lose the dependency

Reply:
Oh, yeah, if that's it then absolutely creating an orphan is the better way to go. If only the scala stdlib defined Monad we could forego these shenanigans!